test/evp_pkey_provided_test.c

```diff
@@ -1717,7 +1732,10 @@ static int test_fromdata_ec(void)
         ret = test_print_key_using_pem(alg, pk)
               && test_print_key_using_encoder(alg, pk);
 
-        if (!ret || !TEST_ptr(dup_pk = EVP_PKEY_dup(pk)))
+        if (!ret || dup_pk != NULL)
+            break;
```

Comment:
this has introduce a memory leak above (which has been caught in the CI failure).

`group, group_p, group_a, group_b` are allocated at lines 1691-1694 and then are clobbered in the second loop iteration.

Reply:
Yep, fixed.